### ISSUE TITLE
Lags in gas estimation 

### ISSUE THREAD

**Issue by michalinacienciala** [tahowallet/extension#3634]
We need to figure out why sometimes estimation of gas seems to be lagging.

_________

From @beemeeupnow:

> I've been noticing that the gas estimate can be 15 to 30 minutes behind in times where gas is increasing.
e.g. gas was ~10 gwei for some time, then steps up to ~19-24. Initiating a transaction after gas has been higher for more than 10 minutes the wallet still suggested ~10 (and submitted max of ~16) 

> just tried another, wallet UI indicated ~9.9 and I clicked Submit, transaction ended up with ~27.9, which is okay since max was set to ~42 based on etherscan 

There was also a situation when sending transaction (on Ethereum Mainnet) resulted in a fee-related error:

```
[2023-09-25T13:28:27.788Z] [DEBUG:BG] SerialFallbackProvider.routeRpcCall
    Skipping fallback for unidentified error {"reason":"processing response error","code":"SERVER_ERROR","body":"{\"jsonrpc\":\"2.0\",\"id\":64,\"error\":{\"code\":-32000,\"message\":\"err: max fee per gas less than block base fee: address [REDACTED], maxFeePerGas: 16432000000 baseFee: 18574135951 (supplied gas 162313)\"}}","error":{"code":-32000},"requestBody":"{\"method\":\"eth_estimateGas\",\"params\":[{\"gas\":\"[REDACTED]\",\"type\":\"[REDACTED]\",\"maxFeePerGas\":\"[REDACTED]\",\"maxPriorityFeePerGas\":\"[REDACTED]\",\"value\":\"[REDACTED]\",\"from\":\"[REDACTED]\",\"to\":\"[REDACTED]\",\"data\":\"[REDACTED]\"}],\"id\":64,\"jsonrpc\":\"2.0\"}","requestMethod":"POST","url":"https://eth-mainnet.alchemyapi.io/v2/REDACTED"} for provider {"_isProvider":true,"_events":[],"_emitted":{"block":18212889,"t:[REDACTED]":18212888},"disableCcipRead":false,"formatter":{"formats":{"transaction":{},"transactionRequest":{},"receiptLog":{},"receipt":{},"block":{},"blockWithTransactions":{},"filter":{},"filterLog":{}}},"anyNetwork":false,"_networkPromise":{},"_maxInternalBlockNumber":18212889,"_lastBlockNumber":18212889,"_maxFilterBlockRange":10,"_pollingInterval":4000,"_fastQueryDate":1695645973348,"connection":{"url":"https://rpc.ankr.com/eth","throttleLimit":1,"timeout":5000},"_nextId":1160,"_eventLoopCache":{"detectNetwork":null},"_pendingBatch":null,"_pendingBatchAggregator":null,"_network":{"name":"homestead","chainId":1,"ensAddress":"[REDACTED]"},"_poller":null,"_bootstrapPoll":null,"_internalBlockNumber":{},"_fastBlockNumber":18212889,"_fastBlockNumberPromise":{}}
```
> after a few minutes, the Activity pane showed the transaction as "Dropped"
etherscan showed it still pending, and it did get confirmed after a few more minutes
definitely room for improvement there since the UI showing something as "Dropped" when still in mempool would confuse those not aware of it

_________

The issue may have similar cause to the problem with `Transaction maxFeePerGas (23907000000) is too low for the next block, which has a baseFeePerGas of 32654986209` errors observed sometimes on environment running on a forked Mainnet.

_________

@andreachapman , could you check if you're able to reproduce this behavior (may not always be possible, as it looks issue can be observed when the gas price is increasing)? You can observe current gas price on Ethereum Mainnet [here](https://etherscan.io/gastracker).
I think it would be good to collect some data on how was the gas price fluctuating and what was the wallet suggesting. When the gas price seems to be underestimated, do transaction and observe if it's successful / if there are any errors.

**Comment by andreachapman** [comment]
@michalinacienciala - I did some testing around this and started a page [here](https://coda.io/d/Build-Team_dznBnP8Wtkc/Lags-in-gas-estimation_suRIL#_lu-W9) for that research. I've been watching it today too but the price hasn't changed much today, nothing like it did yesterday which I would still say was a moderate change in price. 

TLDR:

1. Have observed a lag in what the extension is showing compared to https://etherscan.io/gastracker - though, I'm not sure what lag would be acceptable versus what's too long
2. Did not see any dropped trxns as a result though - I have seen issues where trxns will show as dropped but when I check the scan site they're actually successful. This is captured in a different [issue](https://github.com/tahowallet/extension/issues/2942) where we've been keeping some evidence. But in this particular instance, I didn't see that happening.
